R/correlation.R

```diff
@@ -302,69 +322,65 @@ Correlation <- function(jaspResults, dataset, options){
-        pearson <- pearson$result
+      # even if we do not want the specific tests results
+      # we still want the output as NaN - to fill the jaspTables correctly
+      # so we still loop over all tests - .corr.test() returns empty lists if isFALSE(compute)
+      for(test in c('pearson', 'spearman', 'kendall')){
+        compute <- isFALSE(errors) && .corrTestChecked(test, options)
         
-        spearman <- .corr.test(x = data[,1], y = data[,2], z = condData, 
-                               method = "spearman", alternative = alt, 
-                               conf.level = options$confidenceIntervalsInterval)
-        spearmanErrors <- spearman$errors
-        spearman <- spearman$result
-        
-        kendall <- .corr.test(x = data[,1], y = data[,2], z = condData, 
-                              method = "kendall", alternative = alt, 
-                              conf.level = options$confidenceIntervalsInterval)
-        kendallErrors <- kendall$errors
-        kendall <- kendall$result
-      } else {
-        pearson <- spearman <- kendall <- rep(NaN, length(statsNames))
-        names(pearson) <- names(spearman) <- names(kendall) <- statsNames
-        pearsonErrors <- spearmanErrors <- kendallErrors <- TRUE
+        r <- .corr.test(x = data[,1], y = data[,2], z = condData,
+                        method = test, alternative = alt,
+                        conf.level = options$confidenceIntervalsInterval,
+                        compute = compute, sample.size = currentResults[['sample.size']])
+        testErrors[[test]] <- r[['errors']]
+        currentResults[[test]] <- r[['result']]
       }
-      
       # stolen from manova.R
       shapiro <- .multivariateShapiroComputation(data, list(dependent = .unv(vcomb[[i]])))
       
-      results[[vpair[i]]] <- list(vars = .unv(vcomb[[i]]), vvars = vcomb[[i]], 
-                                  res = list(pearson=pearson, spearman=spearman, kendall=kendall, sample.size = nrow(data)),
-                                  errors = errors, 
-                                  testErrors = list(pearson = pearsonErrors, 
-                                                    spearman = spearmanErrors, 
-                                                    kendall = kendallErrors),
-                                  shapiro = shapiro)
+      results[[vpair[i]]] <- list(vars       = .unv(vcomb[[i]]), 
+                                  vvars      = vcomb[[i]],
+                                  res        = currentResults,
+                                  errors     = errors, 
+                                  testErrors = testErrors,
+                                  shapiro    = shapiro)
       
       # store state for pair
       state <- createJaspState(object = results[[vpair[i]]])
-      state$dependOn(options = c("hypothesis", "confidenceIntervalsInterval", "missingValues"), 
+      state$dependOn(options = c("hypothesis", "confidenceIntervalsInterval", "missingValues",
```

Comment:
Relating to another remark on update json. I dislike our old way of naming arguments such as "confidenceIntervalsInterval". I changed this to ciValue in my code. It's up to you of course on how you'd like to handle this.

Reply:
I am happy to do it, if it's indeed not in a way of backward compatibility, I would gladly change the names that cause me headaches anyway 😄

Reply:
Here's the guide to upgrader: https://github.com/jasp-stats/jasp-desktop/blob/cea6f3cce1966336ee76c75b098901d140e4c8e5/Docs/development/jasp-upgrade-json.md

This upgrader allows you to map old names to new ones. 

An implementation of the upgrader can be found here: https://github.com/jasp-stats/jasp-desktop/blob/cea6f3cce1966336ee76c75b098901d140e4c8e5/Docs/development/jasp-upgrade-json.md

Reply:
How did I not know about this? 😅 This is so cool

Reply:
Haha, yeah and soon it will be in QML so that it doesn't look so complicated :p

Reply:
I decided not to do it for now as I got annoyed having to make the same change twice on two different repos. Maybe it's good idea to make these changes once the whole module approach to R analyses is implemented smoothly... 😄

Reply:
Well you could have just made all the changes in one and only copy it at the last moment ;)
But yeah, I hope to get it all smooth soon...

Reply:
Well yeah, only if I knew every commit I made is not the final one 😉 😄